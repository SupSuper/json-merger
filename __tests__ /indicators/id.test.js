const helpers = require("../../__helpers__");

describe("when merging two objects and a source property has a $id indicator", function () {

    test("it should not merge the $id property", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$id": 1
            }
        };

        const result = helpers.mergeObjects([object1, object2]);

        expect(result).toMatchSnapshot();
    });
});
