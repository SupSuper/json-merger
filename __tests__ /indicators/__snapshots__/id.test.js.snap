// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when merging two objects and a source property has a $id indicator it should not merge the $id property 1`] = `
"{
	\\"a\\": {
		\\"aa\\": \\"original\\"
	}
}"
`;
